src/server/game/Movement/MovementGenerators/TargetedMovementGenerator.cpp

```diff
@@ -237,7 +234,7 @@ bool FollowMovementGenerator<T>::PositionOkay(T* owner, Unit* target, float rang
     if (owner->GetExactDistSq(target) > G3D::square(owner->GetCombatReach() + target->GetCombatReach() + range))
         return false;
 
-    return !owner->IsPet() || !angle || angle->IsAngleOkay(target->GetRelativeAngle(owner));
+    return !owner->IsPet() || !angle || angle->IsAngleOkay(target->GetRelativeAngle(owner)); // need to check - dont think we need !pet exception here because there are scripts with MoveFollow that require angle
```

Comment:
As regards the comment, check the 2nd part of this comment: https://github.com/azerothcore/azerothcore-wotlk/pull/4220#issuecomment-755395084

Multiple guardians summoned by quests and such kind of scripts, they should follow you by an angle specified when you summon them. Another solution would be to check here if the position is free and select a free spot otherwise